rct229/ruleset_functions/compare_schedules.py

```diff
@@ -0,0 +1,68 @@
+from rct229.utils.assertions import RCTFailureException
+
+REGULAR_YEAR_HOUR = 8760
+LEAP_YEAR_HOUR = 8784
+
+
+def compare_schedules(schedule_1: list, schedule_2: list, mask_schedule: list, comparison_factor: float, is_leap_year: bool):
+    """compare two schedules and determine if they match with or without a comparison factor when applicable
+        NOTE: The function only works with hourly schedule for now.
+
+        Parameters
+        ----------
+        schedule_1: list, First schedule [0.187, 0.187, 0.187, ... ]
+        schedule_2: list, Second schedule [0.187, 0.187, 0.187, ...]
+        mask_schedule: list, The schedule that defines comparison mode for all hours (8760 or 8784) in a year, i.e.
+            if hourly value is 1, schedule_1 is evaluated to be equal to schedule_2;
+            if hourly value is 2, schedule_1 is evaluated to be equal to schedule_2 times the comparison factor;
+            if hourly value is 0, comparison was skipped for that particular hour
+            (example when evaluating shut off controls, only he building closed hrs are evaluated) [1,1,1,1,1...]
+        comparison_factor: float, The target multiplier number for schedule_1 compared to schedule_2, i.e. when applicable,
+            the hourly value in schedule_1 shall be equal to that in schedule_2 times the comparison_factor. 1.0
+        is_leap_year: bool, indicate whether the comparison is in a leap year or not. True / False
+
+        Returns
+        -------
+        A dictionary containing the total number of hours that the function compares,
+        the number of hours schedule_1 matches schedule_2 with the comparison_factor,
+        EFLH difference for schedule_1 and schedule_2,
+        i.e. {
+            "TOTAL_HOURS_COMPARED": total_hours_compared,
+            "TOTAL_HOURS_MATCH": total_hours_match,
+            "EFLH_DIFFERENCE: EFLH_difference
```

Comment:
Can you explain what the `elfh_difference` represents?

Reply:
I have this on note for RDS.
By code, it seems the difference should be schedule_1 - schedule_2,
However, the RDS is schedule_1 / schedule_2

Reply:
After review of the RDSs, it seems the elfh_differences is the percentage difference between the two ELFH schedule. It is mostly used in the print statement such as:
`LIGHTING SCHEDULE EFLH IN P-RMR IS ${compare_schedules_result_dictionary[EFLH_DIFFERENCE]} OF THAT IN B-RMR.`

I originally thought the difference is schedule_1 - schedule_2 because standard mentioned 40 hours ELFH differences per week...